src/main/java/org/broadinstitute/hellbender/utils/read/SAMRecordToGATKReadAdapter.java

```diff
@@ -756,6 +754,62 @@ public void reverseComplement() {
         samRecord.reverseComplement(true);
     }
 
+    /**
+     * Clip attributes that change after a hard clipping operation
+     *
+     * The method clips string or array attributes that are part of a pre-set list
+     * originating from FlowBasedRead (a white-list), that are at the same length
+     * as the sequence.
+     */
+    @Override
+    public void hardClipAttributes(final int newStart, final int newLength, final int originalLength)
```

Comment:
@droazen, @jamesemery  - I think this function is useful in general case - it is true that flow based reads added some attributes that should be clipped, but I can see how this could be helpful in a general case. What do you think?

Reply:
@droazen an important detail to note about the flow-based-reads is that they are pretty heavy duty in terms of the extra processing they do so they are mostly lazily instantiated by the tools that need them. This means if we want to do any read processing with flow based reads in most places this has to be generalized.